### ISSUE TITLE
TLA detection still not working (#12986 closed without being fixed)

### ISSUE THREAD

**Issue by perbergland** [meteor/meteor#13299]
Metor 3.0.1

#12986 was closed as fixed but in reality this was never the case.

Reify was fixed but the updated version was not brought in.

#13298 attempts to fix this.

All repro steps in #12986 still apply, e.g. https://github.com/perbergland/tla-repro/tree/main or https://github.com/alisnic/meteor3-tla-repro

**Comment by perbergland** [comment]
It would be pertinent to add unit tests in meteor for this issue and other issues that depend on a combination of internal and external (via npm) code.

I have looked at the self-tests but could not find any suitable place where reify and babel are invoked with the versions specified in the dev-bundle.

**Comment by perbergland** [comment]
Regarding blocking issues with code like

```
if (xxx) {
   const x = require("XXX");
}
```

I wonder if the way TLA was implemented by making require/import async “under the hood” if a package (directly or transitive) has TLA code also makes nested require/import unsupportable? The extra “await” inserted by reify at top level cannot possibly work for nested require/import.

```
if (__reifyWaitForDeps__()) (await __reifyWaitForDeps__())();
```

If my thinking is correct, reify should write out an error and fail if a nested require or import specifies a package that is detected to have TLA code.

**Comment by perbergland** [comment]
For additional context - this is an example of actual emitted code from babel+reify (found in .meteor/local/build/programs/server/app/app.js).
When Reify detects code that has TLA directly of via dependencies, it will set async: true in the loader metadata towards the end.

original code:
```
import axios from "axios";

const getAndrei = async () => {
  const response = await axios.get('https://api.github.com/users/alisnic')
  return response.data
};

const Andrei = await getAndrei();

export default Andrei;
```

Generated code inside app.js:

```
!module.wrapAsync(async function (module, __reifyWaitForDeps__, __reify_async_result__) {
  "use strict";
  try {
    let axios;
    module.link("axios", {
      default(v) {
        axios = v;
      }
    }, 0);
    if (__reifyWaitForDeps__()) (await __reifyWaitForDeps__())();
    const getAndrei = async () => {
      const response = await axios.get('https://api.github.com/users/alisnic');
      return response.data;
    };
    const Andrei = await getAndrei();
    module.exportDefault(Andrei);
    __reify_async_result__();
  } catch (_reifyError) {
    return __reify_async_result__(_reifyError);
  }
  __reify_async_result__()
}, {
  self: this,
  async: false
});
```

**Comment by perbergland** [comment]
Is this planned to be fixed in any upcoming release, or should we resort the workaround of adding e.g. "await 2" in all files that have code that requires TLA to be activated?

**Comment by denihs** [comment]
We want to include a fix for this in the next release, 3.0.4, but we're still investigating this.

**Comment by perbergland** [comment]
Fixed in 3.1
